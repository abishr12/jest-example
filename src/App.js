import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

const Test = () =>  <div>Testing</div>


class App extends Component {
  state = {on: false, input: '',
          mainColor: 'blue'}
  handleStrings(str){
    return true
  }
  componentDidMount(){}
  componentWillReceiveProps(){}
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <ul>
          
            <li>Test 1</li>
            <li>Test 2</li>
            <li>Test 3</li>
          
        </ul>
        <h3 className={this.state.mainColor}>Everyone Okay!</h3>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <p className='button-state'>{
          this.state.on ? 'Yes!' : 'No!'
        }</p>
        <button onClick={() => this.setState({on:true})}>Click</button>
        <h2>{this.state.input}</h2>
        <input onChange={e => this.setState({input: e.target.value})} type ='text'
        />
        <Test />
      </div>
    );
  }
}

export class Link extends Component{
  render(){
    return <a href={this.props.address}>Click</a>
  }
}

export default App;
