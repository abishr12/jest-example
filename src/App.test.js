import React from 'react';
import ReactDOM from 'react-dom';
import App, { Link } from './App';
import {configure, shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json';

configure({adapter: new Adapter()})


describe('<App/> shallow rendering', () => {

  it('ul has 3 li elements', () => {
    const wrapper = shallow(<App/>)
    expect(wrapper.find('ul').children().length).toBe(3)
  })
  it('matches the snapshot', () => {
    const tree = shallow(<App/>)
    expect(toJson(tree)).toMatchSnapshot()
  })
  it('updates className with new State', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find('.blue').length).toBe(1)
    wrapper.setState({mainColor: 'red'})
    expect(wrapper.find('.red').length).toBe(1)
    expect(wrapper.find('.blue').length).toBe(0)
  })
  it('on button click changes p text', () => {
    const wrapper = shallow(<App />)
    const button = wrapper.find('button')
    expect(wrapper.find('.button-state').text()).toBe('No!')
    button.simulate('click')
    expect(wrapper.find('.button-state').text()).toBe('Yes!')
  })
  it('on input change, title changes text', () =>{
    const wrapper = shallow(<App />)
    const input = wrapper.find('input')
    expect(wrapper.find('h2').text()).toBe('')
    input.simulate('change', {target: {value: 'Tyler'}})
    expect(wrapper.find('h2').text()).toBe('Tyler')
  })
  it('calls componentDidMount', () => {
    jest.spyOn(App.prototype,  'componentDidMount')
    const wrapper = shallow(<App />)
    expect(App.prototype.componentDidMount.mock.calls.length).toBe(1)
  })

  it('setProps calls componentWillReceiveProps', () => {
    jest.spyOn(App.prototype, 'componentWillReceiveProps')
    const wrapper = shallow(<App />)
    wrapper.setProps({ hide: true })
    expect(App.prototype.componentWillReceiveProps.mock.calls.length).toBe(1)
    // expect(wrapper.find('.lifeCycle').text()).toBe('componentWillReceiveProps')
  })

  it('handleStrings function returns correctly', () => {
    const wrapper = shallow(<App />)
    const trueReturn = wrapper.instance().handleStrings('Hello World')
    expect(trueReturn).toBe(true)
  })
})

describe('<App/> mount rendering', () => {

  it('ul has 3 li elements', () => {
    const wrapper = mount(<App />)
    expect(wrapper.find('ul').children().length).toBe(3)
    wrapper.unmount()
  })
  it('matches the snapshot', () => {
    const tree = mount(<App />)
    expect(toJson(tree)).toMatchSnapshot()
    tree.unmount()
  })

})

describe('<Link />', () => {

  it('link component accepts prop', () => {

    const wrapper = shallow(<Link address='www.google.com' />)
    expect(wrapper.instance().props.address).toBe('www.google.com')
  })

  it('a tag renders href correctly', () => {

    const wrapper = shallow(<Link address='www.google.com' />)
    expect(wrapper.props().href).toBe('www.google.com')
    
  })
  it('setProps works', () => {

    const wrapper = shallow(<Link address='www.google.com' />)
    wrapper.setProps({ address: 'www.yahoo.com' })
    // console.log(wrapper.get(0))
    expect(wrapper.props().href).toBe('www.yahoo.com')

  })
  
})

// describe('<Link />', () => {
//   const wrapper = shallow(<Link address= 'www.google.com' />)
//   expect(wrapper.instance().props.address).toBe('www.google.com')
// })

// it('renders without crashing', () => { 
//     const div =document.createElement('div');   
//     ReactDOM.render(<App />, div);
// ReactDOM.unmountComponentAtNode(div); });