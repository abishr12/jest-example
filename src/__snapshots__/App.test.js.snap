// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> mount rendering matches the snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
    </header>
    <ul>
      <li>
        Test 1
      </li>
      <li>
        Test 2
      </li>
      <li>
        Test 3
      </li>
    </ul>
    <h3
      className="blue"
    >
      Everyone Okay!
    </h3>
    <p
      className="App-intro"
    >
      To get started, edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <p
      className="button-state"
    >
      No!
    </p>
    <button
      onClick={[Function]}
    >
      Click
    </button>
    <h2 />
    <input
      onChange={[Function]}
      type="text"
    />
    <Test>
      <div>
        Testing
      </div>
    </Test>
  </div>
</App>
`;

exports[`<App/> shallow rendering matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <ul>
    <li>
      Test 1
    </li>
    <li>
      Test 2
    </li>
    <li>
      Test 3
    </li>
  </ul>
  <h3
    className="blue"
  >
    Everyone Okay!
  </h3>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <p
    className="button-state"
  >
    No!
  </p>
  <button
    onClick={[Function]}
  >
    Click
  </button>
  <h2 />
  <input
    onChange={[Function]}
    type="text"
  />
  <Test />
</div>
`;
